.github/scripts/git-filter.py

```diff
@@ -19,14 +19,28 @@ def main():
     compdb = helpers.load_compdb("compile_commands.json")
     compdb = helpers.index_compdb(compdb)
 
+    pdns_path = os.path.join("pdns/", "")
+    cwd = os.getcwd()
     root = helpers.get_repo_root()
 
     diff = sys.stdin.read()
     patch_set = unidiff.PatchSet(diff)
     for patch in patch_set:
         path = os.path.join(root, patch.path)
         if path in compdb:
+            if not patch.path.startswith(pdns_path):
+                # If the file being diffed is not under the pdns/ directory, we
+                # need to reconstruct its filename in the patch adding extra
+                # paths that clang-tidy-diff will get rid of: this way
+                # clang-tidy can work with the correct file path.
```

Comment:
It might be nice to have a small comment with an example of the content of `patch.path` and the subsequent values of `path` and `relpath`. I _think_ I get it but I'm not 100% sure.

Reply:
Done!

Reply:
Thanks a lot! :)